### ISSUE TITLE
Snippets for sh files not working.

### ISSUE THREAD

**Issue by rafamadriz** [hrsh7th/vim-vsnip#182]
Just like the titles says, snippets in sh filetypes aren't working. I've already created the snippets file with `:VsnipOpen` and put all my snippets there, but none of them are showing up.

**Comment by nanotee** [comment]
I've been having this issue as well. Here's how to reproduce it:

1. Install [rafamadriz/friendly-snippets](https://github.com/rafamadriz/friendly-snippets)
2. Install [hrsh7th/vim-vsnip](https://github.com/hrsh7th/vim-vsnip)
3. Install [hrsh7th/nvim-compe](https://github.com/hrsh7th/nvim-compe)
4. nvim-compe configuration:
    ```lua
    require('compe').setup{
        source = {
            vsnip = true,
        },
        debug = true,
    }
    ```
5. Open a `sh` file and start typing
6. Neovim prints `Vim(let):E730: using List as a String`

After spending some time investigating the issue, I found that this line is the cause:

https://github.com/hrsh7th/vim-vsnip/blob/b05641ca8c7ebd396017121219047c480182a743/autoload/vsnip.vim#L183

It assumes `l:snippet.description` is a `String`, but sometimes the dictionary contains a `List` instead.

The error is usually silent because of `nvim-compe`'s `suppress()` function, which is why I had to use debug mode to see it.
